platform/device/udidauth.go

```diff
@@ -94,23 +94,19 @@ func (mw *udidCertAuthMiddleware) Checkin(ctx context.Context, req mdm.CheckinEv
 	if err != nil {
 		return nil, errors.Wrap(err, "error retrieving device certificate")
 	}
-	switch req.Command.MessageType {
-	case "Authenticate":
+	if req.Command.MessageType == "Authenticate" {
 		// unconditionally save the cert hash on Authenticate message
 		if err := mw.store.SaveUDIDCertHash([]byte(req.Command.UDID), hashCertRaw(devcert.Raw)); err != nil {
 			return nil, err
 		}
 		return mw.next.Checkin(ctx, req)
-	case "TokenUpdate", "CheckOut", "GetBootstrapToken", "SetBootstrapToken":
-		matched, err := mw.validateUDIDCertAuth([]byte(req.Command.UDID), hashCertRaw(devcert.Raw))
-		if err != nil {
-			return nil, err
-		}
-		if !matched && !mw.warnOnly {
-			return nil, errors.New("device certifcate UDID mismatch")
-		}
-		return mw.next.Checkin(ctx, req)
-	default:
-		return nil, errors.Errorf("unknown checkin message type %s", req.Command.MessageType)
 	}
+	matched, err := mw.validateUDIDCertAuth([]byte(req.Command.UDID), hashCertRaw(devcert.Raw))
+	if err != nil {
+		return nil, err
+	}
+	if !matched && !mw.warnOnly {
+		return nil, errors.New("device certifcate UDID mismatch")
+	}
+	return mw.next.Checkin(ctx, req)
```

Comment:
You're removing a check for invalid message types here. Just confirming this is because this check also happens in the actual checkin handler (e.g. mw.next)?

Reply:
Yep. For the purposes of UDID<->cert auth we don't really care what the messages are — just that initial authenticate message is handled separately. This removes a duplicate check that has to be maintained whenever messages changed (like they did for BS tokens, and now DM).